pkg/virt-controller/network/pod_annotations.go

```diff
@@ -31,22 +33,43 @@ import (
 )
 
 func GeneratePodAnnotations(networks []virtv1.Network, interfaces []virtv1.Interface, multusStatusAnnotation string, bindingPlugins map[string]virtv1.InterfaceBindingPlugin) map[string]string {
-	newAnnotations := map[string]string{}
+	var err error
+
+	var sriovIfacePciMap map[string]*networkv1.DeviceInfo
 	if vmispec.SRIOVInterfaceExist(interfaces) {
-		networkPCIMapAnnotationValue := deviceinfo.CreateNetworkPCIAnnotationValue(
-			networks, interfaces, multusStatusAnnotation,
-		)
-		newAnnotations[deviceinfo.NetworkPCIMapAnnot] = networkPCIMapAnnotationValue
+		sriovIfaces := vmispec.FilterSRIOVInterfaces(interfaces)
+		sriovIfacePciMap, err = deviceinfo.MapNetworkNameToDeviceInfo(networks, multusStatusAnnotation, sriovIfaces)
+		if err != nil {
+			log.Log.Warningf("failed to create network-device-info-map: %v", err)
+		}
 	}
+
+	var pluginIfacePciMap map[string]*networkv1.DeviceInfo
 	if vmispec.BindingPluginNetworkWithDeviceInfoExist(interfaces, bindingPlugins) {
-		networkDeviceInfoMap, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, multusStatusAnnotation, bindingPlugins)
+		pluginIfacePciMap, err = deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, multusStatusAnnotation, bindingPlugins)
 		if err != nil {
-			log.Log.Warningf("failed to create network-device-info-map: %v", err)
-			networkDeviceInfoMap = map[string]*networkv1.DeviceInfo{}
+			log.Log.Warningf("failed to create network-device-info-map for binding plugin interfaces: %v", err)
 		}
+	}
+	networkDeviceInfoMap := mergeMaps(pluginIfacePciMap, sriovIfacePciMap)
+
+	newAnnotations := map[string]string{}
+
+	if len(networkDeviceInfoMap) > 0 {
 		networkDeviceInfoAnnotation := downwardapi.CreateNetworkInfoAnnotationValue(networkDeviceInfoMap)
 		newAnnotations[downwardapi.NetworkInfoAnnot] = networkDeviceInfoAnnotation
 	}
 
 	return newAnnotations
 }
+
+// mergeNetDeviceInfoMaps create unified map from the given map slice using maps.Copy.
+// When a key of a given map already exist in the unified one, the value in unified map
+// will be overwritten by the value associated with the key in the given map.
+func mergeMaps[M ~map[K]V, K comparable, V any](src ...M) M {
```

Comment:
This generic form is hard to read and not required here. We know exactly what maps we are merging. It is an internal method that is not going to be used anywhere else.

Reply:
I removed the merge-maps because its not longer necessary.